src/comms.cpp

```diff
@@ -117,133 +323,334 @@ void comms_loop() {
-                                }
+                            case GarageDoorCurrentState::CURR_OPENING:
+                                l = "Opening";
+                                break;
+                            case GarageDoorCurrentState::CURR_CLOSED:
+                                l = "Closed";
+                                break;
+                            case GarageDoorCurrentState::CURR_CLOSING:
+                                l = "Closing";
                                 break;
+                            }
+                            RINFO("status DOOR: %s",l);
+
+                            notify_homekit_current_door_state_change();
                         }
-                        if (lock != garage_door.target_lock) {
-                            RINFO("Lock Cmd %d", lock);
-                            garage_door.target_lock = lock;
-                            notify_homekit_target_lock();
+
+                        static GarageDoorTargetState gd_TargetState;
+                        if (garage_door.target_state != gd_TargetState) {
+                            gd_TargetState = garage_door.target_state;
+                            notify_homekit_target_door_state_change();
                         }
-                        // Send a get status to make sure we are in sync
-                        send_get_status();
+
+                        break;
+                    
+                    case 0x39:
                         break;
-                    }
 
-                case PacketCommand::Light:
-                    {
-                        bool l = garage_door.light;
-                        switch (pkt.m_data.value.light.light) {
-                            case LightState::Off:
-                                l = false;
-                                break;
-                            case LightState::On:
-                                l = true;
-                                break;
-                            case LightState::Toggle:
-                            case LightState::Toggle2:
-                                l = !garage_door.light;
-                                break;
+                    // light & lock
+                    case 0x3A:
+
+                        //RINFO("0x3A MSG: %X%02X",key,val);
+
+                        // upper nibble must be 5
+                        if ((val & 0xF0) != 0x50) {
+                            RINFO("0x3A val upper nible not 5: %02X",val);
+                            break;
                         }
-                        if (l != garage_door.light) {
-                            RINFO("Light Cmd %s", l ? "On" : "Off");
-                            garage_door.light = l;
+
+                        lightState = bitRead(val, 2);
+                        lockState  = !bitRead(val, 3);
+
+                        // light status
+                        static uint8_t lastLightState = 0xff;
```

Comment:
I don't think you need this static int.  Can't you just use garage_door.light?  It's essentially a static last light state variable already.

Reply:
Same comment for the lock state

Reply:
maybe, but lock state doesn't really have a bool available (in Garagedoor) to compare for change
so for consistency I want to keep it the same,

Reply:
There is a garage_door.current_lock and a garage_door.target_lock.   The sec+2 logic does the same thing you are doing, except it uses the variables from garage_door rather than declaring additional statics.

Reply:
the vars in garage_door arnt bools for the lock status... so its compare is more complicated
i would have to convert the bool from the opener in to the enum, then look for change
i think code is simpler now with the static var, we got plenty of ram :)

Reply:
The web page code uses and is dependent on `garage_door.current_state` and `current_lock` so this does need to be consistent between protocols.  Yes they are not bools, but we should still use them,  If the 1.0 protocol does not provide the same level of detail, then just use the values that match... possibly just 0 and 1.

Door state:

0. Open
1. Closed
2. Opening
3. Closing
4. Stopped


Lock state:

0. Unsecured
1. Secured
2. Jammed

Reply:
its just 1 byte... i feel it should be like this, as the code is looking for change in the response FROM the GDO, then sets the state of HK based on that